cli/integration/suite_test.go

```diff
@@ -0,0 +1,177 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/log"
+	"github.com/testcontainers/testcontainers-go/modules/registry"
+
+	"ocm.software/open-component-model/bindings/go/oci"
+	urlresolver "ocm.software/open-component-model/bindings/go/oci/resolver/url"
+	"ocm.software/open-component-model/cli/integration/internal"
+)
+
+// TestSuite holds shared test infrastructure
+type TestSuite struct {
+	RegistryAddress string
+	Username        string
+	Password        string
+	ConfigPath      string
+	Repository      *oci.Repository
+}
+
+var globalTestSuite *TestSuite
+
+// TestMain sets up and tears down the test suite
+func TestMain(m *testing.M) {
```

Comment:
Please don't overwrite TestMain. 
I mean i dont like the suite already because it complicates tests, but if you really need one, you can use https://pkg.go.dev/github.com/stretchr/testify/suite#SetupAllSuite for this kind of setup. (please familiarize a bit with the lifecycle methods, there are some more things in there that are useful for you)

Reply:
For me, TestMain is not a great way to initialize stuff because it is global for all tests. Basically as soon as you have 2 suits you have to coinitialize them, and with SetupAllSuite you can split that. I think TestMain should be reserved for initializing all go tests, and not just something specific to a suite.